File: Venmo/src/test/java/com/braintreepayments/api/venmo/VenmoSharedPrefsWriterUnitTest.java
package com.braintreepayments.api.venmo;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

import com.braintreepayments.api.sharedutils.BraintreeSharedPreferences;

import org.junit.Before;
import org.junit.Test;

public class VenmoSharedPrefsWriterUnitTest {

    private BraintreeSharedPreferences braintreeSharedPreferences;

    @Before
    public void beforeEach() {
        braintreeSharedPreferences = mock(BraintreeSharedPreferences.class);
    }

    @Test
    public void persistVenmoVaultOption_persistsVaultOption() {
        VenmoSharedPrefsWriter sut = new VenmoSharedPrefsWriter();
        sut.persistVenmoVaultOption(braintreeSharedPreferences, true);
        verify(braintreeSharedPreferences).putBoolean(
                "com.braintreepayments.api.Venmo.VAULT_VENMO_KEY", true);
    }

    @Test
    public void getVenmoVaultOption_retrievesVaultOptionFromSharedPrefs() {
        VenmoSharedPrefsWriter sut = new VenmoSharedPrefsWriter();
        sut.getVenmoVaultOption(braintreeSharedPreferences);
        verify(braintreeSharedPreferences).getBoolean(
                "com.braintreepayments.api.Venmo.VAULT_VENMO_KEY");
    }
}

File: Venmo/src/test/java/com/braintreepayments/api/venmo/VenmoSharedPrefsWriterUnitTest.kt
package com.braintreepayments.api.venmo

import com.braintreepayments.api.sharedutils.BraintreeSharedPreferences
import io.mockk.mockk
import io.mockk.verify
import kotlin.test.Test

class VenmoSharedPrefsWriterUnitTest {
    private val braintreeSharedPreferences: BraintreeSharedPreferences = mockk(relaxed = true)

    @Test
    fun `persistVenmoVaultOption persists vault option`() {
        val sut = VenmoSharedPrefsWriter()
        sut.persistVenmoVaultOption(braintreeSharedPreferences, true)
        verify { braintreeSharedPreferences.putBoolean("com.braintreepayments.api.Venmo.VAULT_VENMO_KEY", true) }
    }

    @Test
    fun `getVenmoVaultOption retrieves vault option from SharedPrefs`() {
        val sut = VenmoSharedPrefsWriter()
        sut.getVenmoVaultOption(braintreeSharedPreferences)
        verify { braintreeSharedPreferences.getBoolean("com.braintreepayments.api.Venmo.VAULT_VENMO_KEY") }
    }
}
